spring/src/main/java/io/seata/spring/annotation/GlobalTransactionScanner.java

```diff
@@ -246,6 +258,20 @@ protected Object wrapIfNecessary(Object bean, String beanName, Object cacheKey)
         }
     }
 
+    private int getAddAdvisorPos(AdvisedSupport advised, Advisor avr, int avrOrder) {
+        Integer order;
+        Advisor advisor;
+        for (int i = 0, l = advised.getAdvisors().length; i < l; ++i) {
+            advisor = advised.getAdvisors()[i];
+            order = OrderUtil.getOrder(advisor);
+            if (order == null || order > avrOrder ||
+                    (order == avrOrder && avr.getClass().getSimpleName().compareTo(advisor.getClass().getSimpleName()) != 1)) {
+                return i;
+            }
+        }
+        return advised.getAdvisors().length - 1;
```

Comment:
`advised.getAdvisors().length - 1` => `advised.getAdvisors().length` ?

Reply:
No, this method returns the pos to `advised.addAdvisor(pos, avr);`, it needs to be `-1`.

Reply:
I've rewritten the code here, and it should be better understood now.
PTAL